[CDAP-18322] Refactoring: Move artifact sidecar cache prewarming from TaskWorkerService to ArtifactLocalizerService

wyzhang opened the pull request:
Why:
ArtifactLocalizer sidecar pattern can be enabled for both task worker
and preview runner, moving artifact cache prewarming to ArtifactLocalizerService
avoid duplicating the logic in preview runner and task worker service.

masoud-io commented on cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/k8s/SparkContainerDriverLauncher.java:
```diff
@@ -271,11 +272,11 @@ private static Location createBundle(Path workingDirectory) throws IOException {
     ArtifactLocalizerClient(DiscoveryServiceClient discoveryServiceClient,
                             InternalAuthenticator internalAuthenticator,
-                            CConfiguration cConf) {
+                            CConfiguration cConf, ArtifactManagerFactory artifactManagerFactory) {
```
nit: would you please move cConf to the beginning.

wyzhang replied:
Will do a separate PR.

masoud-io reviewed (approved):
LGTM. 
Note that ArtifactLocalizerServiceTest tests are failing.

wyzhang committed:
[CDAP-18322] Refactoring: Move artifact sidecar cache prewarming from TaskWorkerService to ArtifactLocalizerService

Why:
ArtifactLocalizer sidecar pattern can be enabled for both task worker
and preview runner, moving artifact cache prewarming to ArtifactLocalizerService
avoid duplicating the logic in preview runner and task worker service.
